README.md

```diff
@@ -1,2 +1,310 @@
-# validator
-A tiny runtime library for type assertions
+# Validator
+
+A tiny library for run time type assertions.
+
```

Comment:
Since there's a lot of stuff here, do you think it would make sense to have a table of contents? remark-preset-davidtheclark can help with this — you could look at the remark usage in one of the repos that does this, if you're interested.

Reply:
For now, I added a simple table of contents.